src/proto_reader/reflection_fetch.rs

```diff
@@ -0,0 +1,307 @@
+    let descriptor = protox_parse::parse("reflection", REFLECTION_PROTO)?;
+    let mut descriptor_set = FileDescriptorSet::default();
+    descriptor_set.file.push(descriptor);
+    ProtobufSet::from_proto_file(descriptor_set)
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Service {
+    name: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ListServicesResponse {
+    service: Vec<Service>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct FileDescriptorProtoResponse {
+    file_descriptor_proto: Vec<String>,
+}
+
+impl FileDescriptorProtoResponse {
+    fn get(self) -> Result<Vec<u8>> {
+        let file_descriptor_proto = self
+            .file_descriptor_proto
+            .first()
+            .context("Received empty fileDescriptorProto")?;
+
+        BASE64_STANDARD
+            .decode(file_descriptor_proto)
+            .context("Failed to decode fileDescriptorProto from BASE64")
+    }
+}
+
+/// Used for serializing all kinds of GRPC Reflection responses
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CustomResponse {
+    list_services_response: Option<ListServicesResponse>,
+    file_descriptor_response: Option<FileDescriptorProtoResponse>,
+}
+
+impl CustomResponse {
+    async fn execute(
+        url: &str,
+        grpc_method: GrpcMethod,
+        body: serde_json::Value,
+        target_runtime: &TargetRuntime,
+    ) -> Result<CustomResponse> {
+        let protobuf_set = get_protobuf_set()?;
+        let reflection_service = protobuf_set.find_service(&grpc_method)?;
+        let operation = reflection_service.find_operation(&grpc_method)?;
+        let mut url: url::Url = url.parse()?;
+        url.set_path(
+            format!(
+                "{}.{}/{}",
+                grpc_method.package, grpc_method.service, grpc_method.name
+            )
+            .as_str(),
+        );
+        let req_template = RequestTemplate {
+            url: Mustache::parse(url.as_str())?,
+            headers: vec![(
+                HeaderName::from_static("content-type"),
+                Mustache::parse("application/grpc+proto")?,
+            )],
+            body: Mustache::parse(body.to_string().as_str()).ok(),
+            operation: operation.clone(),
+            operation_type: Default::default(),
+        };
+
+        let ctx = ConfigReaderContext {
+            runtime: target_runtime,
+            vars: &Default::default(),
+            headers: Default::default(),
+        };
+
+        let req = req_template.render(&ctx)?.to_request()?;
+
+        let resp = target_runtime.http.execute(req).await?;
+        let body = resp.body.as_bytes();
+
+        let response: CustomResponse = operation.convert_output(body)?;
+        Ok(response)
+    }
+}
+
+/// Makes `ListService` request to the grpc reflection server
+pub async fn list_all_files(url: &str, target_runtime: &TargetRuntime) -> Result<Vec<String>> {
```

Comment:
can we collect all of the functions related to reflection in single struct and its impl?

It will simplify passing the same arguments everywhere and not calculate same things again.